### PR TITLE
Restore submodules to dmlc

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
Restore submodules to dmlc
Please also merge:

https://github.com/dmlc/mshadow/pull/313
https://github.com/dmlc/dmlc-core/pull/345

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by larroy** [comment]
@cjolivier01

File: .gitmodules [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [submodule "mshadow"]
 	path = mshadow
-	url = https://github.com/larroy/mshadow.git
+	url = https://github.com/dmlc/mshadow.git
```
Comment (eric-haibin-lin) [inline .gitmodules]: Is this change sufficient to change the submodule? Do you have to do `git add mshadow`/ `git add dmlc-core` to specify a particular commit id for each one of them?

Comment (larroy) [reply .gitmodules]: Yes they have to be updated. See the related PRS linked in the description please.

**Comment by rahul003** [comment]
Can you make the build pass? (I think master is still protected). 
I guess unless CI starts from a fresh clone it won't have your submodules cloned, so it will be unable to checkout your commit.

**Comment by larroy** [comment]
I need dependent PRs merged!

https://github.com/dmlc/mshadow/pull/313
https://github.com/dmlc/dmlc-core/pull/345

Can you guys please read the descriptions on the PRs?

**Comment by rahul003** [comment]
Ya I understand. I was referring to whether all builds will get stuck now that the reference to your fork has gone in, and CI can't checkout that commit. I asked because an already cloned repository I have can't checkout build because it can't find your commit in dmlc repo. But it looks like the CI does a fresh clone each time, so it's fine. Never mind.

**Comment by larroy** [comment]
yes when the subrepos change you have to sync them again

**Pull request merged**